classes/boardmanager.php

```diff
@@ -0,0 +1,1152 @@
+                $newcolumn[$column->id]->id = $DB->insert_record('kanban_column', $newcolumn[$column->id]);
+            }
+            foreach ($cards as $card) {
+                $newcard[$card->id] = clone $card;
+                $newcard[$card->id]->kanban_board = $newboard['id'];
+                $newcard[$card->id]->timecreated = time();
+                $newcard[$card->id]->timemodified = time();
+                $newcard[$card->id]->kanban_column = $newcolumn[$card->kanban_column]->id;
+                $newcard[$card->id]->originalid = $card->id;
+                unset($newcard[$card->id]->id);
+                // Remove user id of original creator.
+                unset($newcard[$card->id]->createdby);
+                $newcard[$card->id]->id = $DB->insert_record('kanban_card', $newcard[$card->id]);
+                // Copy attachment files.
+                if ($context) {
+                    $this->copy_attachment_files($context->id, $card->id, $newcard[$card->id]->id);
+                }
+            }
+
+            $newboard['sequence'] = helper::sequence_replace($newboard['sequence'], $newcolumn);
+            $DB->update_record('kanban_board', $newboard);
+            helper::update_cached_board($newboard['id']);
+            foreach ($newcolumn as $col) {
+                $col->sequence = helper::sequence_replace($col->sequence, $newcard);
+                $DB->update_record('kanban_column', $col);
+            }
+            return $newboard['id'];
+        }
+    }
+
+    /**
+     * Deletes a board and all contents of it.
+     *
+     * @param int $id The board id
+     * @return void
+     */
+    public function delete_board(int $id) {
+        global $DB;
+        // Cards need to be read to identify files, assignees and discussions.
+        $cardids = $DB->get_fieldset_select('kanban_card', 'id', 'kanban_board = :id', ['id' => $id]);
+        $this->delete_cards($cardids);
+
+        $DB->delete_records('kanban_history', ['kanban_board' => $id]);
+        $DB->delete_records('kanban_column', ['kanban_board' => $id]);
+        $DB->delete_records('kanban_card', ['kanban_board' => $id]);
+        $DB->delete_records('kanban_board', ['id' => $id]);
+        // The rest of the elements is skipped in the update message.
+        $this->load_board($id);
+        $this->formatter->delete('board', ['id' => $id]);
+    }
+
+    /**
+     * Delete multiple cards and all attached data (discussions, assignees, files, calendar events).
+     *
+     * @param array $ids The card ids
+     * @param bool $updatecolumn Whether to update the column sequence (can be set to false, if column is going to be deleted)
+     * @return void
+     */
+    public function delete_cards(array $ids, bool $updatecolumn = true): void {
+        foreach ($ids as $id) {
+            $this->delete_card($id, $updatecolumn);
+        }
+    }
+
+    /**
+     * Delete a card and all attached data (discussions, assignees, files, calendar events).
```

Comment:
also history?

Reply:
Not now - history could also reflect deleted cards